tools/quickstart/quickstart/cli.py

```diff
@@ -1,110 +1,25 @@
-        # parity can't handle
-        json_account = account.encrypt(password, kdf="pbkdf2")
-
-        os.makedirs(os.path.dirname(os.path.abspath(self.keystore_file)), exist_ok=True)
-        with open(self.keystore_file, "w") as f:
-            f.write(json.dumps(json_account))
-
-        with open(self.password_file, "w") as f:
-            f.write(password)
-
-        with open(self.address_file, "w") as f:
-            f.write(account.address)
-
-
-def is_wrong_password_error(err):
-    return type(err) is ValueError and str(err) == "MAC mismatch"
-
-
-def import_keystore_file(trustline_files, keystore_input_file):
-    keyfile_dict = json.loads(open(keystore_input_file, "rb").read())
-
-    while 1:
-        password = click.prompt(text="Password", hide_input=True)
-        try:
-            account = Account.from_key(Account.decrypt(keyfile_dict, password))
-            break
-        except Exception as err:
-            if is_wrong_password_error(err):
-                click.echo("The password you entered is wrong. Please try again.")
-            else:
-                click.echo(f"Error: malformed keystore file: {repr(err)}")
-                sys.exit(1)
-    trustline_files.store(account, password)
-    sys.exit(0)
-
-
-def read_private_key():
-    while 1:
-        privkey = click.prompt(text="Private key (hex encodded)", hide_input=True)
-        if re.fullmatch("[0-9a-fA-F]{64}", privkey):
-            return privkey
-        click.echo(
-            "The private key muss be entered as hex encoded string. Please try again."
-        )
-
-
-def read_password():
-    while 1:
-        password = click.prompt(text="Password", hide_input=True)
-        password2 = click.prompt(text="Password (repeat)", hide_input=True)
-        if password == password2:
-            return password
-        click.echo("Passwords do not match. Please try again.")
-
-
-def import_private_key(trustline_files):
-    private_key = read_private_key()
-    account = Account.from_key(private_key)
-    click.echo(f"Read private key for address {account.address}")
-    password = read_password()
-
-    trustline_files.store(account, password)
-
-
-def make_trustlines_files():
-    password_file, address_file, keystore_file = sys.argv[1:4]
-    return TrustlinesFiles(
-        password_file=password_file,
-        address_file=address_file,
-        keystore_file=keystore_file,
-    )
 
+from quickstart import bridge, docker, netstats, validator_account
 
-def qs_import_keystore_file():
-    if len(sys.argv) != 5:
-        click.echo(
-            "Usage: qs-import-keystore-file PASSWORD_FILE ADDRESS_FILE KEYSTORE_FILE KEYSTORE_INPUT_FILE"
-        )
-        sys.exit(1)
-    import_keystore_file(make_trustlines_files(), keystore_input_file=sys.argv[4])
 
+@click.command()
+@click.option(
+    "--host-base-dir",
+    help=(
+        "absolute path to use for docker volumes "
+        "(only relevant when run from inside a docker container itself)"
```

Comment:
No, it is especially important for them, because they use it like this in 99% of all cases.
So how can we improve this... Any suggestions? What's not clear? :thinking:

Reply:
users will just run something like `./quickstart` on their machine, right? So not within a docker container. So I don't know what you want to say with the sentence in parentheses.

Reply:
Nope. The quickstart script starts the docker container which handles the setup. The intention by Ralf and me to have no requirements for the user. Therefore Python (dependencies) and anything else is available for the user by guarantee.

Reply:
> Can we remove this option?

No we can't. Without, the tool does not work from inside Docker anymore. Because it is the primary target to run within a container, this is a support we can not drop.

A common user will not see this help text, because he never starts the tool directly on his own, but use the quickstart script (differ between tool and script). Else the option description is correct.

**Background:**
The _docker-compose_ configuration can't use absolute paths for volumes here, because how to know where exactly on each machine? So in difference to _Docker_ it allows to define relative paths, which it will then expand. This does not work correctly, when running `docker-compose` via the quickstart tool inside its own container. The trick is, that bounding the _Docker_ socket to the quickstart container lets you control _Docker_ on the host and spawn new containers etc. So a container orchestrates other containers. But you can not spwan a new container with a volume reference to you current containers virtual filesystem. That must be the external path on the host. And that's exactly what this option is about. It can't determine this information else without a huge bunch of overhead by parsing its own container configuration.

Reply:
I see, I didn't realize this wasn't a user facing interface.